### DISCUSSION TITLE
blog/w/taichi-ray-tracing

### DISCUSSION THREAD

**Discussion by giscus** [HK-SHAO/HK-SHAO.github.io#28] [General]
# blog/w/taichi-ray-tracing

本文是追光小队为 Taichi Hackathon 2022 所写; GitHub 仓库在 https://github.com/HK-SHAO/RayTracingPBR; 在线效果演示 raytracing.shao.fun (https://raytracing.shao.fun); 四天极限开发，爆肝写文，笔者也只是业余爱好者，本文可能会有诸多...

https://shao.fun/blog/w/taichi-ray-tracing.html

**Comment by WeebOwO** [comment]
建立坐标系这一节的图片描述是不是有误，图中不应该是右手坐标系么？😢

**Reply by HK-SHAO** [reply]
没错的话，是以XZ为地面的左手系呀？

**Reply by WeebOwO** [reply]
正常来讲不会说以xz为参考吧，我看很多的书籍都将作者图片中的情况描述为右手坐标系，以xy为基准，按照右手定则指出z轴的方向。
![](https://p1.ssl.qhimg.com/t010a79ade73278d27a.jpg)

**Reply by HK-SHAO** [reply]
我查看了 Wikipedia 这张图 ( https://zh.wikipedia.org/wiki/File:3D_Cartesian_Coodinate_Handedness.jpg )，确定了本文确实应该是右手系，这个我一直也搞错了😂

**Reply by HK-SHAO** [reply]
你好，这是必应。😊

计算机图形学中，没有一个统一的标准来建立场景的世界空间坐标系，不同的软件或者库可能会使用不同的手系。一般来说，左手系和右手系的区别在于z轴的方向，如果x轴和y轴的方向相同，那么z轴的方向就是两个手系的区别。¹²

左手系和右手系的判断方法是，如果用右手（左手）的大拇指指向第一个坐标轴（x轴）的正向，而其余手指以第二个轴（y轴）绕第一轴转动的方向握紧，就与第三个轴（z轴）重合，就称此坐标系为右手（左手）坐标系。²

一些常用的左手系的软件或者库有Direct3D，Unity，3ds Max等，一些常用的右手系的软件或者库有OpenGL，Maya，Blender等。³⁴

左手系和右手系的优劣没有绝对的说法，主要取决于个人的习惯和使用的工具。一般来说，左手系更符合人类的视觉习惯，因为人类的眼睛在头部的左右两侧，所以看到的景象的深度是从左到右增加的，而右手系更符合数学的习惯，因为右手系的z轴是从屏幕向外的，而不是向里的。¹

Source: Conversation with Bing, 15/2/2023(1) 左手系与右手系_hhjian6666的博客-CSDN博客_左手系. https://blog.csdn.net/qq_30796379/article/details/79837032 Accessed 15/2/2023.
(2) 右手系・左手系とは ～ 定義と例 ～ - 理数アラカルト. http://www.risalc.info/src/right-handed-left-handed.html Accessed 15/2/2023.
(3) 左手坐标系和右手坐标系详解_zsffuture的博客-CSDN博客_右手坐标系. https://blog.csdn.net/weixin_42398658/article/details/106054760 Accessed 15/2/2023.
(4) 从左手坐标系到右手坐标系的变换_LoongTech的博客-CSDN博客_左手坐标系和右手坐标系怎么转换. https://blog.csdn.net/loongkingwhat/article/details/88528702 Accessed 15/2/2023.
